frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { sortable } from '@patternfly/react-table';
+
+import {
+  TemplateSource,
+  getTemplateOperatingSystems,
+  getTemplateFlavors,
+  TEMPLATE_TYPE_LABEL,
+} from 'kubevirt-web-ui-components';
+
+import { ListPage, Table, TableRow, TableData } from '@console/internal/components/factory';
+import { Kebab, ResourceLink, ResourceKebab } from '@console/internal/components/utils';
+import { getName, getNamespace, DASH } from '@console/shared';
+import { TemplateModel } from '@console/internal/models';
+
+export const menuActions = Kebab.factory.common;
+
+const { kind } = TemplateModel;
+const selector = {
+  matchLabels: { [TEMPLATE_TYPE_LABEL]: 'vm' },
+};
+const labelPlural = 'Virtual Machine Templates';
+
+const tableColumnClass = classNames('col-lg-2', 'col-sm-4', 'col-xs-4');
+const tableColumnClassHiddenOnSmall = classNames('col-lg-2', 'hidden-sm', 'hidden-xs');
+
+const tableColumnClasses = [
+  tableColumnClass,
+  tableColumnClass,
+  tableColumnClass,
+  tableColumnClassHiddenOnSmall,
+  tableColumnClassHiddenOnSmall,
+  tableColumnClassHiddenOnSmall,
+  Kebab.columnClass,
+];
+
+const VmTempleateTableHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Namespace',
+      sortField: 'metadata.namespace',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Description',
+      sortField: 'metadata.annotations.description',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Source',
+      props: { className: tableColumnClasses[3] },
+    },
+    {
+      title: 'OS',
+      props: { className: tableColumnClasses[4] },
+    },
+    {
+      title: 'Flavor',
+      props: { className: tableColumnClasses[5] },
+    },
+    {
+      title: '',
+      props: { className: tableColumnClasses[6] },
+    },
+  ];
+};
+VmTempleateTableHeader.displayName = 'VmTemplateTableHeader';
+
+const VmTemplateTableRow = ({ obj: template, index, key, style }) => {
+  const os = getTemplateOperatingSystems([template])[0];
+
+  return (
+    <TableRow id={template.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink
+          kind={kind}
+          name={getName(template)}
+          namespace={getNamespace(template)}
+          title={template.metadata.uid}
```

Comment:
`getUid` is available from `@console/shared` as of today.

Reply:
yeee :-)